src/plugins/data/public/ui/typeahead/suggestions_component.tsx

```diff
@@ -29,12 +29,56 @@ interface Props {
   show: boolean;
   suggestions: QuerySuggestion[];
   loadMore: () => void;
+  queryBarRect?: DOMRect;
+  dropdownHeight?: string;
 }
 
 export class SuggestionsComponent extends Component<Props> {
   private childNodes: HTMLDivElement[] = [];
   private parentNode: HTMLDivElement | null = null;
 
+  kbnTypeaheadDivRefInstance: RefObject<HTMLDivElement> = createRef();
+  kbnTypeaheadPopoverDivRefInstance: RefObject<HTMLDivElement> = createRef();
+
+  updatePosition = () => {
+    let suggestionsListHeight;
+    const kbnTypeaheadPopoverDiv = this.kbnTypeaheadPopoverDivRefInstance.current;
+    if (kbnTypeaheadPopoverDiv) {
+      kbnTypeaheadPopoverDiv.style.maxHeight = this.props.dropdownHeight
+        ? this.props.dropdownHeight
+        : '60vh';
+      suggestionsListHeight = kbnTypeaheadPopoverDiv.clientHeight;
+    }
+
+    const documentHeight = document.documentElement.clientHeight || window.innerHeight;
+    const kbnTypeaheadDiv = this.kbnTypeaheadDivRefInstance.current;
+
+    if (this.props.queryBarRect) {
+      const {
```

Comment:
I don't see any benefits of creating that constants. 

```
const { queryBarRect } = this.props;  
queryBarRect.left (17 chars)
queryBarPositionLeft (20 chars)
```
+ it saves at least 5 lines